include/rapidjson/document.h

```diff
@@ -700,8 +700,10 @@ class GenericValue {
             return StringEqual(rhs);
 
         case kNumberType:
-            if (IsDouble() || rhs.IsDouble())
-                return GetDouble() == rhs.GetDouble(); // May convert one operand from integer to double.
+            if (IsDouble() || rhs.IsDouble()) {
+                float delta = GetDouble() - rhs.GetDouble();    // May convert one operand from integer to double.

```

Comment:
Thank you. I mistakenly write `float` instead of `double` as well. But since GetDouble() is not just returning a value and may involve conversion, I do this instead:
~~~cpp
                double a = GetDouble();     // May convert from integer to double.
                double b = rhs.GetDouble(); // Ditto
                return a >= b && a <= b;    // Prevent -Wfloat-equal
~~~
I applied De Morgan's law :)